### PR TITLE
feat: add `__floordiv__` and `__rfloordiv__` for pyarrow

### PR DESCRIPTION
**Pull request by lucianosrp**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #543 #372  #491 
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.
PR splitted to give more space for truediv

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
thanks @lucianosrp 

I think this may need adjusting for negative numbers:
```

In [6]: pl.Series([-100]) // pl.Series([48])
Out[6]:
shape: (1,)
Series: '' [i64]
[
        -3
]

In [7]: pc.divide(-100, 48)
Out[7]: <pyarrow.Int64Scalar: -2>
```

**Comment by MarcoGorelli** [comment]
could just copy (with a link) what pandas does? https://github.com/pandas-dev/pandas/blob/262fcfbffcee5c3116e86a951d8b693f90411e68/pandas/core/arrays/arrow/array.py#L124-L154

**Review by MarcoGorelli — APPROVED** [review]
thanks @lucianosrp !

small thing: in a recent PR, we renamed `tests/expr` to `tests/expr_or_series`, and `tests/series` to `tests/series_only`, so that's causing a conflict

if that's resolved, the rest looks good 👍

**Comment by lucianosrp** [comment]
@MarcoGorelli , I think I did a bit of a mess, let me know if this looks fine 😅

**Comment by MarcoGorelli** [comment]
thanks @lucianosrp ! the code looks good, it just looks like it fails on old versions of pyarrow

I think all we need to do is, in `__floordiv__`, to do a check on `self._backend_version`, and if it's below a certain version (to be determined), then we raise `"__floordiv__ requires at least pyarrow version ..."`?

File: tests/series_only/arithmetic_test.py [hunk 1]
```diff
@@ -34,11 +42,45 @@ def test_arithmetic(
...
+        nw.col("a") // right
+    )
+    compare_dicts(result, expected)
+    if parse_version(pd.__version__) >= (2, 2):
+        # Bug in old version of pandas
+        result = nw.from_native(
+            pd.DataFrame({"a": [left]}).convert_dtypes(dtype_backend="pyarrow"),
+            eager_only=True,
+        ).select(nw.col("a") // right)
+        compare_dicts(result, expected)
```
Comment (MarcoGorelli) [inline tests/series_only/arithmetic_test.py:74]:
```suggestion
    if parse_version(pd.__version__) >= (2, 2):
        # Bug in old version of pandas
        result = nw.from_native(
            pd.DataFrame({"a": [left]}).convert_dtypes(dtype_backend="pyarrow"),
            eager_only=True,
        ).select(nw.col("a") // right)
        compare_dicts(result, expected)
    else:  # pragma: no cover
        pass
```

**Pull request merged**